ppa/archive/views.py

```diff
@@ -166,3 +171,72 @@ def render_to_csv(self, data):
 
     def get(self, *args, **kwargs):
         return self.render_to_csv(self.get_data())
+
+
+class AddToCollection(ListView, FormMixin, ProcessFormView):
+    '''
+    View to bulk add a queryset of :class:`ppa.archive.models.DigitizedWork`
+    to a set of :class:`ppa.archive.models.Collection instances`.
+
+    Restricted to staff users via staff_member_required on url.
+    '''
+
+    model = DigitizedWork
+    template_name = 'archive/add_to_collection.html'
+    form_class = AddToCollectionForm
+    success_url = reverse_lazy('admin:archive_digitizedwork_changelist')
+
+    def get_queryset(self, *args, **kwargs):
+        '''Return a queryset filtered by id, or empty list if no ids'''
+        # get ids from session if there are any
+        ids = self.request.session.get('collection-add-ids', [])
+        # if somehow a problematic non-pk is pushed, will be ignored in filter
+        digworks = DigitizedWork.objects.filter(id__in=ids if ids else [])
+        # revise the stored list in session
+        self.request.session.ids = list(digworks)
```

Comment:
I'm not quite following the view logic for why you have to put the ids back in the session.  Is this because you need the list on GET for display and then on POST for processing?  Maybe clarify this in the comments a bit?